_template_python/INDICATOR_DEV_GUIDE.md

```diff
@@ -355,25 +355,136 @@ Next, the `acquisition.covidcast` component of the `delphi-epidata` codebase doe
       12. `value_updated_timestamp`: now
    2. Update the `epimetric_latest` table with any new keys or new versions of existing keys.
 
+### CI/CD
+
+* Add module name to the `build` job in `.github/workflows/python-ci.yml`.
+  This allows github actions to run on this indicator code, which includes unit tests and linting.
+* Add top-level directory name to `indicator_list` in `Jenkinsfile`.
+  This allows your code to be automatically deployed to staging after your branch is merged to main, and deployed to prod after `cocivcast-indicators` is released.
+* Create `ansible/templates/{top_level_directory_name}-params-prod.json.j2` based on your `params.json.template` with some adjustment:
+   * "export_dir": "/common/covidcast/receiving/{data-source-name}"
+   * "log_filename": "/var/log/indicators/{top_level_directory_name}.log"
+
+Pay attention to the receiving/export directory, as well as how you can store credentials in vault.
+Refer to [this guide](https://docs.google.com/document/d/1Bbuvtoxowt7x2_8USx_JY-yTo-Av3oAFlhyG-vXGG-c/edit#heading=h.8kkoy8sx3t7f) for more vault info.
+
 ### Staging
 
-After developing the pipeline code, but before deploying in development, the pipeline should be run on staging for at least a week. This involves setting up some cronicle jobs as follows:
+After developing the pipeline code, but before release to prod, the pipeline should be run on staging for at least a week.
 
-first the indicator run
+The indicator run code is automatically deployed on staging after your branch is merged into main. After merging, make sure you have proper access to Cronicle and staging server `app-mono-dev-01.delphi.cmu.edu` _and_ can see your code on staging at `/home/indicators/runtime/`.
 
-Then the acquisition run
+Then, on Cronicle, create two jobs: First one to run the indicator and second one to load the output csv files into database.
 
-See [@korlaxxalrok](https://www.github.com/korlaxxalrok) or [@minhkhul](https://www.github.com/minhkhul) for more information.
+#### Indicator run job
+
+This job ssh into our staging server and run the indicator, producing csv files output.
+
+Example script:
 
-https://cronicle-prod-01.delphi.cmu.edu/#Schedule?sub=edit_event&id=elr5clgy6rs
+```
+#!/bin/sh
+
+# vars
+user='automation'
+host='app-mono-dev-01.delphi.cmu.edu'
```

Comment:
hostname publicly available [here](https://github.com/cmu-delphi/www-epicast/blob/f84314885760ea8d54a62401fe182a86e0e87e4e/docs/epicast_development.md?plain=1#L66) and [here](https://github.com/cmu-delphi/covidcast-indicators/blob/1526dc940f61e5a1ab8ff2ce795f35c965c2e48e/ansible/inventory#L6). User potentially derivable [here](https://github.com/search?q=repo%3Acmu-delphi%2Foperations%20user%20automation&type=code).

Reply:
@korlaxxalrok which of these should be removed? Do we need to change any user or host names?

Reply:
It doesnt hurt to scrub as much of the sensitive stuff as possible...  Even if some bits of it are already "out there", reducing the number of current instances shrinks the attack surface (kinda) and makes it a little harder for a malicious actor to find useful information.

I definitely prefer having documentation like this in version control -- its easier to find, use, and see the timeline of changes.  It puts instructions and details "next to" the code assets that are being discussed, and in a format closer to whats practical for developers and operations (as opposed to the complicated wysiwyg editor of gdocs).

We could create a new private documentation-only repository or even just a top-level `/documentation` directory in the existing private https://github.com/cmu-delphi/delphi-admin repo.  It wouldnt be as nice as having the docs in the same repo as the code, but it would be protected and cross-references will be handled well by the github web ui (for privileged users).

Reply:
ie: 
* https://github.com/cmu-delphi/delphi-admin/issues/204

Reply:
Something like [Docsify](https://docsify.js.org/#/) might be nice, for eventually building a docs site from Markdown files. I'd maybe push for a separate repo, to make it simpler to clone the repo without getting extra stuff (delph-admin comes with _a lot_ of extra bits), to add access control, and to eventually make it easier to add CI to build a site from it.

We tried something like this before, but with a different tool (mkdocs), ultimately building it into a small site at https://delphi.cmu.edu/systems-handbook. There is a process to clone the repo, make changes, build it locally for review, and publish changes. It never caught on though, and the wheel kept going round.

Reply:
I agree completely, great points. The "where" is definitely a problem. I wanted to put the "how to make an indicator" manual in this repo so it's close to all the indicator code.

Alternative ideas:

- Put documentation/manuals in relevant repo, but omit host/user/login details and just say "talk to Brian"
- Ditto but link to login details stored in a separate private repo
- Put all documentation/manuals in a dedicated private repo. I like that it'd be on GH, but harder to discover for a given topic
- Put all documentation/manuals in a dedicated folder in GDrive. Also not discoverable

Reply:
@melange396 [opened an issue](https://github.com/cmu-delphi/delphi-admin/issues/204) for this. `delphi-admin` is private and so a good place to store sensitive info.